sota-client/docker/sota_provision.sh

```diff
@@ -56,18 +61,19 @@ register_device() {
 }
 
 register_ecus() {
-  echo "Generating RSA keypair for Primary ECU"
+  echo "Registering ECUs with Director..."
   openssl genpkey -algorithm RSA -out "$out_pri.der" -outform DER -pkeyopt rsa_keygen_bits:2048
   openssl rsa -pubout -in "$out_pri.der" -inform DER -out "$out_pri.pub"
   pubkey=$(sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/\\n/g' < "$out_pri.pub")
   ecus='{"ecu_serial":"'"$primary_serial"'","hardware_identifier":"'"$hardware_id"'","clientKey":{"keytype":"RSA","keyval":{"public":"'"$pubkey"'"}}}'
 
-  echo "Reading secondary ECUs from $cert_dir/$in_ecus"
-  while read -r secondary; do
-    [[ "$secondary" =~ [^\{.*\}$] ]] && ecus+=",$secondary"
+  while read -r serial hw_id; do
+    openssl genpkey -algorithm RSA -out "$serial.der" -outform DER -pkeyopt rsa_keygen_bits:2048
+    openssl rsa -pubout -in "$serial.der" -inform DER -out "$serial.pub"
+    pubkey=$(sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/\\n/g' < "$serial.pub")
```

Comment:
What are you doing here? And can you make a note for when people look at this in the future.